add higher percision to discount, and referral fee perentages in ROUX…

gratbodkin opened the pull request:
…_ReferralBasedCrowdsale

remove account address from truffle-config.js

commented out code in test/helpers/decodeLogs.js because it was causing errors when testing due to missing dependency (likely a web3 versioning issue of some kind)

gbodkin committed:
add higher percision to discount, and referral fee perentages in ROUX_ReferralBasedCrowdsale

remove account address from truffle-config.js

commented out code in test/helpers/decodeLogs.js because it was causing errors when testing due to missing dependency (likely a web3 versioning issue of some kind)

piersmana commented on contracts/crowdsale/ROUX_RefferalBasedCrowdsale.sol:
```diff
@@ -24,9 +24,12 @@ contract ROUX_ReferralBasedCrowdsale is MintedCrowdsale, PostDeliveryCrowdsale,
-    uint256 mMaxDiscount = 20;
-    uint256 oneHundred = 100;
+    // Maximum allowable referral reward percentage (i.e. 300 = 3%)
+    uint256 mMaxReferralPercentageFee = 300;
```
Can we have a default instead of maximum?

gratbodkin replied:
sounds good

piersmana commented on contracts/crowdsale/ROUX_RefferalBasedCrowdsale.sol:
```diff
@@ -24,9 +24,12 @@ contract ROUX_ReferralBasedCrowdsale is MintedCrowdsale, PostDeliveryCrowdsale,
+    // Maximum allowable referral reward percentage (i.e. 300 = 3%)
+    uint256 mMaxReferralPercentageFee = 300;
+    // Maximum allowable token discount percentage (i.e. 2000 = 20%)
+    uint256 mMaxDiscount = 2000;
```
Let's set the maximum to 10000 (100%) and the default to 2000.

gratbodkin replied:
souds good

gratbodkin replied:
dont you think we should have some of maximum that serves as a safe gaurd?

gratbodkin replied:
I would rather not give it any default, unless the default is 0

piersmana commented:
I think the defaults should be zeroed out… And I agree, there should be a
cap on the referral bonus. Perhaps hundred

On Apr 8, 2018 10:43 PM, "T. Grattan Bookin" <notifications@github.com>
wrote:

*@gratbodkin* commented on this pull request.
------------------------------

In contracts/crowdsale/ROUX_RefferalBasedCrowdsale.sol
<https://github.com/roux-io/roux/pull/1#discussion_r179991600>:

> @@ -24,9 +24,12 @@ contract ROUX_ReferralBasedCrowdsale is MintedCrowdsale, PostDeliveryCrowdsale,

     mapping(address => Backer) mBackers;

-    uint256 mMaxReferralPercentageFee = 3;
-    uint256 mMaxDiscount = 20;
-    uint256 oneHundred = 100;
+    // Maximum allowable referral reward percentage (i.e. 300 = 3%)
+    uint256 mMaxReferralPercentageFee = 300;
+    // Maximum allowable token discount percentage (i.e. 2000 = 20%)
+    uint256 mMaxDiscount = 2000;

I would rather not give it any default

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub
<https://github.com/roux-io/roux/pull/1#discussion_r179991600>, or mute the
thread
<https://github.com/notifications/unsubscribe-auth/AFYWcNOU44zf2AbhRHN1J39JaUbnqccvks5tmvT1gaJpZM4TLt_t>
.

gbodkin committed:
change maximum percentage for discounts, and referral fees to be 100%, with no default value.

added boolean to enable easily turning on/off the deployments listed in deployments.js

NOTES:
Attempting to add a referral source account without providing a discount, and referral rate will result in the transaction being reverted.

piersmana commented:
:+1:
